### DISCUSSION TITLE
Welcome to WebLaTex Discussions!

### DISCUSSION THREAD

**Discussion by sanjib-sen** [sanjib-sen/WebLaTex#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by aravindhnivas** [comment]
Hii, 

Thanks for this great project.
Just a quick question: No PDF is generated after saving the document and also there is no Latex compiler in terminal->output.

I have forked it and used it in codespace.

Am I missing something ?

**Reply by sanjib-sen** [reply]
Hi, thank you Aravindh for using this tool.  Can you delete the fork and fork again / use as a template to create a new instance and check if it is working again or not?

You can also click on "use as a template" button to create a new instance / repo / without deleting the previous fork. 

Then retest it. Please keep it in mind that,
When you save for the first time, the compiler takes 1-2 minutes to compile for the first time. Then it is instant. 
If there are any error, you can check the Error outputs in the "Latex workshop" tab of the terminal. Not output. Or check the latex workshop extension.

Also, sometimes you need to change the compiler to work in certain latex documents. My thesis supervisor used 'lualatex' as a compiler for some specific tools, there is also 'xelatex'. You can try changing the compiler too with a single comment in the root latex file.
More on this:
https://github.com/sanjib-sen/WebLaTex#to-use-with-lualatex-or-any-other-tex-program

So, let me know if any of these works. You can also try this tool offline with your vscode and debug everything using docker. I am saying this because it is hard to come up with a solution without your latex files. But that could break some privacy, so...

**Comment by lkoehl** [comment]
I noticed that you are using the [docker-texlive](https://github.com/dante-ev/docker-texlive) repository. This uses an older version of texlive, but adds some additional features such as pandoc. Are the extra features needed or would it be helpful to upgrade to the [official texlive docker container](https://hub.docker.com/r/texlive/texlive) which provides the latest version?

**Reply by sanjib-sen** [reply]
Thank you for your suggestion. The main purpose of this project was to combine texlive and GitHub together. The official texlive docker container does not have git or any other program installed. So, I had to use a container where both texlive and git are installed. There was another problem. The official texlive container is so large (almost 8GB to download) and not optimized at all. It also has a lot of libraries installed which may not be needed by someone. So, it took GitHub Codespace 4-5 mins to install the containers in the first time. I have tried to use an optimized docker container which does not take much time and have git installed.

However, I am also looking for other containers and trying different containers. I will happily change this if a latest one which is also optimized and have git installed comes in front. I am also trying to make a docker container by myself with necessary software installed and optimization. I have made two of them, which are still under development. Hopefully will be able to use them/ one of it in the near future.

You can also test out other docker containers, just change the container name in the devcontainer.json and see how they work. Let me know if you found a better one.

**Reply by lkoehl** [reply]
I tried the official image. Am I seeing it correctly that the build time is a one time thing when the codespace is created? If so, I would not focus on reducing the build time that much. Personally, I would prefer a newer version of texlive if it comes with a slightly longer installation time. Having git installed is a great thing though.

**Reply by sanjib-sen** [reply]
> I tried the official image. Am I seeing it correctly that the build time is a one time thing when the codespace is created? If so, I would not focus on reducing the build time that much. Personally, I would prefer a newer version of texlive if it comes with a slightly longer installation time. Having git installed is a great thing though.

You are right. Anyway, I will continue researching ways to optimize the official docker-image and add features like git so we can use that on `WebLaTex` in the meantime, I can provide you another version of WebLaTex with the latest and full `texlive` support. Are you interested?

**Comment by lkoehl** [comment]
Since I am fairly new to codespaces and docker, I would like to know if it would be possible to make `.tex` files accessible from other folders. In my personal workflow, I have created several packages and document classes that I use. They are all stored in a style folder. On my local machine, I just put all the files in the TeX home directory. Is there a way to do the same with WebLaTex?

A little bit more visualized: 
```
weblatex/
├─ style/
│  ├─ my_class.cls
├─ PDF/
│  ├─ demo.pdf
├─ files/
│  ├─ demo.tex
```

**Reply by sanjib-sen** [reply]
> Since I am fairly new to codespaces and docker, I would like to know if it would be possible to make `.tex` files accessible from other folders. In my personal workflow, I have created several packages and document classes that I use. They are all stored in a style folder. On my local machine, I just put all the files in the TeX home directory. Is there a way to do the same with WebLaTex?
> 
> A little bit more visualized:
> 
> ```
> weblatex/
> ├─ style/
> │  ├─ my_class.cls
> ├─ PDF/
> │  ├─ demo.pdf
> ├─ files/
> │  ├─ demo.tex
> ```

Sorry for the late reply. The answer is `Yes`. It is definitely possible to make `.tex` files accessible from other folders. I mean this is the main reason for using `LaTex`. Why there would not be?
